packages/fx-core/src/plugins/resource/apiconnector/plugin.ts

```diff
@@ -34,15 +42,88 @@ export class ApiConnectorImpl {
     const projectPath = inputs.projectPath;
     const languageType: string = ctx.projectSetting!.programmingLanguage!;
     const config: ApiConnectorConfiguration = this.getUserDataFromInputs(inputs);
-    for (const componentItem of config.ComponentPath) {
-      await this.scaffoldEnvFileToComponent(projectPath, config, componentItem);
-      await this.scaffoldSampleCodeToComponent(projectPath, config, componentItem, languageType);
-      // await this.addSDKDependency(ComponentPath);
-    }
+    // backup relative files.
+    const backupFolderName = generateTempFolder();
+    await Promise.all(
+      config.ComponentPath.map(async (component) => {
+        await this.backupExistingFiles(path.join(projectPath, component), backupFolderName);
+      })
+    );
 
+    try {
+      await Promise.all(
+        config.ComponentPath.map(async (component) => {
+          await this.scaffoldInComponent(projectPath, component, config, languageType);
+        })
+      );
+    } catch (err) {
```

Comment:
minor case: the err caught maybe a systemError/userError type, which can be used directly.

Reply:
thank you, will resolve in the next pr